docs/integration-testing.md

```diff
@@ -25,8 +25,8 @@ eventually we might want to write an integration test that uses a CosmosDB test
 
 ## Coding Guidelines
 
-An integration test is annotated with `@IntegrationTest`, which causes the test runner to ignore it unless the
-`RUN_INTEGRATION_TEST` environment variable is set to `true`.
+An integration test should have an annotation e.g.`@IntegrationTest, @AzureCosmosDbIntegrationTest` which causes the test runner to ignore it unless the
+`RUN_INTEGRATION_TEST` environment variable is set to `true` and also categorize them using [Junit Tags](https://junit.org/junit5/docs/current/user-guide/#writing-tests-tagging-and-filtering). This categorization is important because it allows us discover and executed specific set of integration tests.
```

Comment:
why do we need both a property and an environment variable? Seems overly complex.

Reply:
Here environment variable is used to by default not execute integration tests. (This is already used in EDC). I thought of removing this environment variable  but then we have the following situation. 

- We need one common tag (or list of tags) for Junit excludeTags() option which prevents these tests from executing by default.
- If we go with one common tag approach then for executing any specific IT test required additional params. For comparison currently to execute a particular test in minicube.sh we do 
`./gradlew :samples:04.0-file-transfer:integration-tests:test --tests org.eclipse.dataspaceconnector.samples.FileTransferAsClientIntegrationTest`  but then we have to always apply the include tags options also, which mean above command will become

`./gradlew :samples:04.0-file-transfer:integration-tests:test -DincludeTags="integration-test" --tests org.eclipse.dataspaceconnector.samples.FileTransferAsClientIntegrationTest`
(also same for IDE you need to always pass specific -DincludeTags while executing different IT tests.)


As EDC community is already have a set process to use RUN_INTEGRATION_TEST environment variable so I thought of reusing it and built upon it.

Reply:
Sure, but why couldn't it be yet another tag?

for minikube, create a specific tag ust for that test.

I don't think the EDC community is attached to a way of running tests, they just need one defined :)

Reply:
Sure, we can have a tag for that as well. But I think `RUN_INTEGRATION_TEST `environment variable also brings additional convivence for devs , for example imagine if a dev wants to run unit+ITs test for azure blob-provision module then with help of this environment variable devs can just set `RUN_INTEGRATION_TEST=true` and execute 
`./gradlew -p extensions/azure/blob-provision test` and this will execute unit+ITs for them.

But if we have default exclude tags(for the reasons mentioned above) then you always have to specify -DincludeTags and executing unit+ITs seems not possible as we do not have tags for UTs.

Reply:
You could configure the task to (optionally) run non-tagged tests too?

if (includeTags.isNotEmpty()) {
...
}
else if (System.getProperty("RunUnitAndIntegrationTest")){
// do not include anythign
}
...

Reply:
Updated the approach without use of RUN_INTEGRATION_TEST and also added a way to execute all of types. Please take a look.